app/assets/javascripts/s3_file_field.js.coffee

```diff
@@ -71,8 +71,16 @@ jQuery.fn.S3FileField = (options) ->
 
     formData: (form) ->
       unique_id = @files[0].unique_id
+      file_key = $this.data('key').replace('{timestamp}', new Date().getTime()).replace('{unique_id}', unique_id)
+
+      # assume ${filename} is present and can be prefixed with a ms_timestamp
+      ms_timestamp = new Date().getTime()
```

Comment:
I guess there's no negative affect of having drift between this value and the `new Date().getTime()` on line 74. They don't need to be identical, and also would technically help prevent collisions if they in fact are different. Is that correct?

Reply:
Yes! But the pathname isn't used on the server side processing, as we use Rails for file processing, which uses `File.basename`.